Codecademy/gamut
# chore(deps): bump marked from 0.7.0 to 4.3.0

### Overview



This pull request updates marked to version ^4.3.0 to fix current security vulnerabilities. The `parseInline()` function is used in place of `inlineLexer()`. 

Also deployed on portal and in the le in this [pr](https://github.com/codecademy-engineering/mono/pull/2819).

Please note in order for this to pass tests in monorepo, I needed to set the `mangle: false` in the markedOptions, in packages/gamut/src/Markdown/index.tsx. E

```
 const markedOptions = {
      smartypants: true,
      headerIds,
      headerPrefix: 'heading-',
      mangle: false,
    };
```
Mangle is mentioned in the documentation [here](https://marked.js.org/using_advanced) and is deprecated in the most recent version of marked in favor of the marked-mangle package. Mangling is used for mailto links to prevent email harvesting bots from collecting email addresses from websites. Currently, we are not mangling emails on prod. 



### PR Checklist


#### Testing Instructions



[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

1. In the pr preview, ensure that markdown and inline markdown appears the same as it does on production. 
2. See how this version appears in portal and the LE in this pr: https://github.com/codecademy-engineering/mono/pull/2778

<img width="500" alt="Screen Shot 2023-07-18 at 3 55 16 PM" src="https://github.com/Codecademy/gamut/assets/4481281/0bff36ac-529f-423a-b44a-4ffd7dee47fe">


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |
